src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyPairGenerator.java

```diff
@@ -151,7 +151,7 @@ public void initialize(int keySize, SecureRandom random) {
         try {
             checkKeySize(keySize, null);
         } catch (InvalidAlgorithmParameterException e) {
-            throw (InvalidParameterException) new InvalidParameterException(e.getMessage()).initCause(e);
+            throw new InvalidParameterException(e.getMessage(), e);
```

Comment:
Here too I would change this to `throw new InvalidParameterException(e)` since the detail message of the cause will be used, the first parameter doesn't add any value.